opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -778,19 +790,20 @@ def use_span(
                 context_api.detach(token)
 
         except Exception as error:  # pylint: disable=broad-except
-            if (
-                isinstance(span, Span)
-                and span.status is None
-                and span._set_status_on_exception  # pylint:disable=protected-access  # noqa
-            ):
-                span.set_status(
-                    Status(
-                        canonical_code=StatusCanonicalCode.UNKNOWN,
-                        description="{}: {}".format(
-                            type(error).__name__, error
-                        ),
+            # pylint:disable=protected-access
+            if isinstance(span, Span):
+                if span._record_exception:
```

Comment:
I think we should re-examine the approach of how to set the flag though. It feels weird to me that the span needs to have a flag set to determine if an exception should be recorded or not, solely for it's usage in the use_span method.

Upon examination, I think the _set_status_on_exception flag should also be fixed in the same fashion.

Both _set_status_on_exception and record_exception here only change the behavior of use_span, which is an operation *on* the object, rather than an operation of the object itself.

I think having "record_exception" as an attribute on the span sets the false exception that the span somehow knows how to capture exceptions in situations where it's the active, despite that being the responsibility of use_span.

I would argue for:

- start_as_current_span, use_span both having a record_exception, set_status_on_exception flag
- start_span does not, as it's just responsible for creating a span.
- remove the "_set_status_on_exception / _record_exception" fields. 

For this pr, I would say it's sufficient just to not use this antipattern with record_exception.

Reply:
Makes a lot of sense. This is how I actually had the first version implemented but then changed it to be more consistent with the "status" feature. Thanks for bringing this up.